lib/deref/index.js

```diff
@@ -1,61 +1,57 @@
-var Rx = require("rx/dist/rx");
-var pathSyntax = require("falcor-path-syntax");
+var __key = require("./../internal/key");
+var __path = require("./../internal/path");
+var __parent = require("./../internal/parent");
+var InvalidDerefInputError = require("./../errors/InvalidDerefInputError");
 
-module.exports = function deref(boundPathArg) {
+module.exports = function deref(boundJSONArg) {
+    var path = boundJSONArg && boundJSONArg[__path];
+    var key = boundJSONArg && boundJSONArg[__key];
 
-    var model = this;
-    var modelRoot = model._root;
-    var pathsIndex = -1;
-    var pathsCount = arguments.length - 1;
-    var paths = new Array(pathsCount);
-
-    var boundPath = pathSyntax.fromPath(boundPathArg);
-
-    while (++pathsIndex < pathsCount) {
-        paths[pathsIndex] = pathSyntax.fromPath(arguments[pathsIndex + 1]);
+    // This is an invalid argument.  Must* be a generated JSON object from
+    // get.  * - Technically this can just be passed in but don't do that.
+    if (path === undefined && key === undefined) {
+        throw new InvalidDerefInputError();
     }
 
-    if (modelRoot.syncRefCount <= 0 && pathsCount === 0) {
-        throw new Error("Model#deref requires at least one value path.");
+    // The simple deref when there is a path.
+    if (path) {
+        return this._clone({
```

Comment:
There may be some things we can do to make clone faster but that is orthogonal to this PR.  I would like to measure that first, to confirm suspicions, before proceeding with optimizations.